src/main/scala/diplomacy/Nodes.scala

```diff
@@ -10,82 +10,258 @@ import freechips.rocketchip.util.HeterogeneousBag
+  * It contains such phases:
+  *   1. Node binding(non-lazy)
+  *     `nodeB := nodeA` is a typical node binding in diplomacy.
+  *     It means `nodeA` as master [[OutwardNode]], `nodeB` as slave [[InwardNode]] will be connected together.
+  *     when executing `nodeB := nodeA`, `nodeB` will invoke [[InwardNode.bind]], which is implemented with [[MixedNode.bind]]:
+  *       `nodeA` will call `nodeA.oPush(index, node, binding)`, in [[OutwardNode.oPush]]:
+  *         `index` is current `nodeA` [[OutwardNode.accPO]] size, which represents the how many [[InwardNode]] has connected to `nodeA`
+  *         `node` is [[NodeHandle]] of `NodeB`,
+  *         `binding` is binding type, which is used for deciding port mapping behavior.
+  *       similarly, `nodeB`, will call [[InwardNode.iPush]] after `nodeA` finish its binding.
+  *   2. Port mapping(lazy)
+  *     Port mapping converts [[OutwardNode.accPO]] and [[InwardNode.accPI]] to [[MixedNode.oPortMapping]] and [[MixedNode.iPortMapping]]
+  *     it is implemented with [[MixedNode.resolveStar]], which solves the size of each connection.
+  *   3. Parameter negotiation(lazy)
+  *     After finishing port mapping,
+  *     [[MixedNode.doParams]] and [[MixedNode.uiParams]] are result of parameter negotiation,
+  *   4. Bundle building.(lazy)
+  *     After finishing parameter propagation,
+  *     [[MixedNode.edges]] can be generated by zipping ports and parameters together.
+  *
+  * Diplomacy has some implicit acronym described below:
+  *   Parameters System:
+  *   D[IO], U[IO], E[IO], B[IO] is parameters will be propagated.
+  *   D: Downwards (master -> slave)
+  *   U: Upwards (slave -> master)
+  *   E: Edge contains necessary of nearby nodes connection information.
+  *   B: Bundle is [[chisel3.Data]], which you can attach chisel data type to it.
+  *
+  * */
+
+/** enable [[InwardNodeImp.monitor]],
+  * which is used by [[freechips.rocketchip.tilelink.TLMonitorBase]] to generate Monitor Module.
+  * */
 case object MonitorsEnabled extends Field[Boolean](true)
-case object RenderFlipped extends Field[Boolean](false)
 
+/** flip edge illustrated in the GraphML. */
+case object RenderFlipped extends Field[Boolean](false)
+/** [[RenderedEdge]] can set the color and label of the generated DAG graph. */
 case class RenderedEdge(
   colour:  String,
   label:   String  = "",
-  flipped: Boolean = false) // prefer to draw the arrow pointing the opposite direction of other edges
-
-// DI = Downwards flowing Parameters received on the inner side of the node
-// UI = Upwards   flowing Parameters generated by the inner side of the node
-// EI = Edge Parameters describing a connection on the inner side of the node
-// BI = Bundle type used when connecting to the inner side of the node
+  flipped: Boolean = false)
+
+/** [[InwardNodeImp]] is the Slave Interface implementation
```

Comment:
> Can we split the "Nodes.scala" PR separate from the other one? I think Nodes is super important, but shouldn't necessarily block the other.

Sure I can create another PR for `Nodes.scala`, since It is unfinished version, and I also failed to understand `resolveStar` :(

> I feel like this explicitly avoided "master/slave" naming

What's your suggestion? I think we should name some conception firstly.